### PR TITLE
[Misc] Simplify block quantization logic

### PR DESCRIPTION
**Pull request by kyuyeunk**
# Description

Remove padding option for quantization and perform padding outside of quantization logic.

# Tests

Please describe how you tested this change, and include any instructions and/or
commands to reproduce.

# Checklist

Before submitting this PR, please make sure:
- I have performed a self-review of my code.
- I have necessary comments in my code, particularly in hard-to-understand areas.
- I have made or will make corresponding changes to any relevant documentation.

### CODE REVIEW

File: tpu_inference/layers/common/quantization.py [hunk 1]
```diff
@@ -239,16 +226,15 @@ def quantize_tensor(
 
     abs_max = jnp.max(jnp.abs(tensor), axis=axis, keepdims=True)
     scale = abs_max / dtype_max
+
+    # If scale=0, scale_inv=1/scale=1/0=NaN. Since NaN will cause numeric error
```
Comment (vanbasten23) [inline tpu_inference/layers/common/quantization.py:230]: thanks for the comment!

File: tpu_inference/layers/common/quantization.py [hunk 2]
```diff
@@ -191,35 +189,24 @@ def quantize_tensor(
         axis = [axis]
```
Comment (vanbasten23) [inline tpu_inference/layers/common/quantization.py:190]: Do you need to check the precondition that the padding has been done before this function is called?

Comment (kyuyeunk) [reply tpu_inference/layers/common/quantization.py:190]: We already have a check for htat. in line 200, we have a raise `ValueError`. so if a dim size doesn't properly align to block size - which potentially mean that a tensor wasn't properly padded.
